client/src/features/LogIn/components/LogInForm.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-unused-vars */
+import React, { useState } from 'react';
+import { Box, Button, Typography, CircularProgress, Grid } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { TextField } from 'formik-mui';
+import { Formik, Field, Form } from 'formik';
+import * as yup from 'yup';
+import { useDispatch } from 'react-redux';
+import postLogIn from '../../../api/postLogIn';
+import { toggleIsLogin } from '../../../store/isLogin';
+
+const buttonSX = {
+  width: { xs: '86px', mobile: '110px', tablet: '140px', laptop: '150px' },
+  px: 0,
+  py: { xs: '8px', mobile: '12px', desktop: '14px' },
+  mx: 4,
+  fontSize: { xs: '12px', laptop: '14px' },
+};
+
+const preloaderSX = {
+  my: 5,
+  width: '80px !important',
+  height: '80pX !important',
+};
+
+const gridItemSx = {
+  width: { xs: '80%', tablet: '60%', laptop: '40%' },
+  my: { xs: '4%', tablet: '3%' },
+};
+
+const LogInForm = () => {
+  const [errorMassage, setErrorMassage] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const myHandleSubmit = async (values) => {
+    setIsLoading(true);
+
+    const response = await postLogIn(values).then((res) => res);
+
+    if (response.success) {
+      localStorage.setItem('token', response.token);
+      setErrorMassage(null);
+      navigate('/');
+      dispatch(toggleIsLogin());
+    } else {
+      setErrorMassage(response?.password || response?.loginOrEmail);
+    }
+
+    setIsLoading(false);
```

Comment:
unfortunately this is incorrect,   we are using Redux store manager, all logic and data are encapsulated and contained in the store, so baded on that, all await async functions, are written within the reducer using the THUNK middleware logic built in.

Reply:
this will be fixed in the future